src/jdk.hotspot.agent/macosx/native/libsaproc/ps_core.c

```diff
@@ -32,8 +32,14 @@
 #include "libproc_impl.h"
 #include "ps_core_common.h"
 
-#ifdef __APPLE__
+#define UNSUPPORTED_ARCH "Unsupported architecture!"
+
+#if defined(__x86_64__)
 #include "sun_jvm_hotspot_debugger_amd64_AMD64ThreadContext.h"
+#elif defined(__arm64__)
+#include "sun_jvm_hotspot_debugger_aarch64_AARCH64ThreadContext.h"
+#else
+#error UNSUPPORTED_ARCH
 #endif
```

Comment:
can we make it uniformly with MacosxDebugfferLocal.m here please?

using amd64 define instead of __x86_64__

arm64 instead of __arm64__